tests/installation/system_role.pm

```diff
@@ -51,16 +51,13 @@ sub assert_system_role {
     # Still initializing the system at this point, can take some time
     # Asserting screen with preselected role
     # Proper default role assertion will be addressed in poo#37504
-    assert_screen 'system-role-default-system', 180;
-    my $system_role = get_var('SYSTEM_ROLE', 'default');
-    if (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default')) {
-        change_system_role($system_role);
-    }
+    assert_screen [qw(before-role-selection system-role-default-system)], 180;
+    my $system_role = is_opensuse ? get_var('SYSTEM_ROLE') : get_var('SYSTEM_ROLE', 'default');
+    change_system_role($system_role) if (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default'));
```

Comment:
Just a though, as we don't use advantage of multiple assert screen, so in case of `before-role-selection` match, we may report that SYSTEM_ROLE is not defined. WDYT?